File: 1988-MinimizeMaximumPairSumInArray/1988-MinimizeMaximumPairSumInArray.py
# Last updated: 7/9/2026, 9:47:50 AM
class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        nums.sort()
        n=len(nums)
        ans=0
        for i in range(n):
            a=nums[i]+nums[n-i-1]
            ans=max(ans,a)
        return ans

File: 1988-MinimizeMaximumPairSumInArray/1988-MinimizeMaximumPairSumInArray.java
// Last updated: 7/9/2026, 9:47:56 AM
class Solution {
    public int minPairSum(int[] nums) {
        Arrays.sort(nums);
        int sum=0;
        int l=0;
        int r=nums.length-1;
        int max= Integer.MIN_VALUE;
        while(l<r){
            sum=nums[l]+nums[r];
            r--;
            l++;
            max=Math.max(max,sum);
            
            
        }
       
        return max;
    }
}